### PR TITLE
Add handling for UpdateAutoTimestamp when not in a transaction

### PR DESCRIPTION
**Pull request by CydeWeys**
It's not clear why this is sometimes causing test flakes, but getting better
logging involved should help clear it up.

---
This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/1341)

### CODE REVIEW

**Review by gbrodman — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @CydeWeys and @gbrodman)

___
*[core/src/test/java/google/registry/testing/DatabaseHelper.java, line 1207 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOWkgR6aNcRtchlLdE:-MkOWkgS7cif4TrZPADf:b-2jhto8) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/test/java/google/registry/testing/DatabaseHelper.java#L1207)):*
> ```Java
>    * ForeignKeyedEppResources.
>    */
>   public static <R> ImmutableList<R> persistSimpleResources(final Iterable<R> resources) {
> ```

Does this cause any issue with any possible situations where the caller of `persistSimpleResources` is persisting something for which a version has already been persisted? I'd imagine that that could possibly result in errors with the `tm().insertAll()` calls

**Review by CydeWeys — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @gbrodman)

___
*[core/src/test/java/google/registry/testing/DatabaseHelper.java, line 1207 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOWkgR6aNcRtchlLdE:-MkOXNTE9vUvJGTNeANu:b6amt45) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/test/java/google/registry/testing/DatabaseHelper.java#L1207)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

Does this cause any issue with any possible situations where the caller of `persistSimpleResources` is persisting something for which a version has already been persisted? I'd imagine that that could possibly result in errors with the `tm().insertAll()` calls
</blockquote></details>

None that are immediately obvious (`persistSimpleResources()` isn't used that widely).  And if it did cause any tests to fail, just fix those tests!  All the callsites I looked at were persisting things for the first time.

**Review by gbrodman — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @CydeWeys)

___
*[core/src/test/java/google/registry/testing/DatabaseHelper.java, line 1207 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOWkgR6aNcRtchlLdE:-MkOY6Mj5HcUUDS6Go0s:b-mtzqx) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/test/java/google/registry/testing/DatabaseHelper.java#L1207)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

None that are immediately obvious (`persistSimpleResources()` isn't used that widely).  And if it did cause any tests to fail, just fix those tests!  All the callsites I looked at were persisting things for the first time.
</blockquote></details>

fair enough

**Review by gbrodman — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @CydeWeys and @gbrodman)

___
*[core/src/main/java/google/registry/model/UpdateAutoTimestamp.java, line 74 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOZ_Q8E1S0Ini_ygEU:-MkOZ_Q8E1S0Ini_ygEV:b-y8h6xd) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/main/java/google/registry/model/UpdateAutoTimestamp.java#L74)):*
> ```Java
>     }
>     if (autoUpdateEnabled() || lastUpdateTime == null) {
>       timestamp = jpaTm().getTransactionTime();
> ```

we should keep setting the timestamp for comparison purposes

**Review by CydeWeys — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @gbrodman)

___
*[core/src/main/java/google/registry/model/UpdateAutoTimestamp.java, line 74 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOZ_Q8E1S0Ini_ygEU:-MkO_-0mECdcnWUDJReW:b6czevz) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/main/java/google/registry/model/UpdateAutoTimestamp.java#L74)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

we should keep setting the timestamp for comparison purposes 
</blockquote></details>

Already done.

**Review by gbrodman — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @CydeWeys and @gbrodman)

___
*[core/src/main/java/google/registry/model/UpdateAutoTimestamp.java, line 74 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOZ_Q8E1S0Ini_ygEU:-MkO_ID21LeAQwEJftQb:b8h7irn) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/main/java/google/registry/model/UpdateAutoTimestamp.java#L74)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

Already done.
</blockquote></details>

it doesn't look like it, I'm referring to setting the `timestamp` object

**Review by CydeWeys — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @gbrodman)

___
*[core/src/main/java/google/registry/model/UpdateAutoTimestamp.java, line 74 at r1](https://reviewable.io/reviews/google/nomulus/1341#-MkOZ_Q8E1S0Ini_ygEU:-MkObVt43kfJlYEP8joE:brmt4rv) ([raw file](https://github.com/google/nomulus/blob/edcea01e2cb7e657055fec2fdebeeea03b6e20e1/core/src/main/java/google/registry/model/UpdateAutoTimestamp.java#L74)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

it doesn't look like it, I'm referring to setting the `timestamp` object
</blockquote></details>

Whoops, my bad.  Didn't realize that `timestamp` was itself a field and not just a local variable.  Set it too.

**Review by gbrodman — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1341)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @CydeWeys)

File: core/src/test/java/google/registry/model/registrar/RegistrarTest.java [hunk 1]
```diff
@@ -270,8 +270,8 @@ void testSuccess_emptyContactTypesAllowed() {
     persistSimpleResource(
         new RegistrarContact.Builder()
             .setParent(registrar)
-            .setName("John Abused")
-            .setEmailAddress("johnabuse@example.com")
+            .setName("John Abussy")
```
Comment (gbrodman) [inline core/src/test/java/google/registry/model/registrar/RegistrarTest.java:273]: What was the reason for this change?

**Pull request merged**